docs/dataset/core-concepts/dataset.md

```diff
@@ -30,10 +30,39 @@ In essence, datapoints in this context are versatile, immutable data units that
 
 Structure your dataset as a *CSV file*. Each row in the file should represent a distinct datapoint.
 
-- **Mandatory Columns**: for models that process images, audio, or video, include a **locator** column with valid URL paths to the respective files. For text-based models, include a **text** column that contains the input text data directly in the CSV.
+- **Reserved Columns**: for models that process images, audio, or video, include a `locator` column with valid URL paths to the respective files. For text-based models, include a `text` column that contains the input text data directly in the CSV.
 
-- **Additional Fields**: Include relevant metadata depending on the data type. For instance, image datasets might have metadata like 'image_width', 'image_height', etc. Similarly, other data types can have their respective metadata fields that are useful for model processing.
+- **Additional Fields**: Include relevant metadata depending on the data type. For instance, image datasets might have metadata like `image_width`, `image_height`, etc. Similarly, other data types can have their respective metadata fields that are useful for model processing.
 
 - **Data Consistency and Format**: It's crucial to maintain data consistency. URLs should be correctly encoded, text should be properly formatted, and numerical values should adhere to their respective formats.
 
 - **Data Accessibility**: Ensure the data, especially if linked through URLs, is accessible for processing. In the case of cloud storage, appropriate permissions should be in place to allow access.
+
+### How to shape a CSV file for ingestion
+
+#### For multimedia datapoints
+
+Add a `locator` column in the CSV to view multimedia datapoints in the web application.
+
+#### For text-focused datapoints
+
+Add a `text` column in the CSV to view text datapoints in the web application.
+
+#### To incorporate [`annotation`](../../reference/annotation.md) and [`asset`](../../reference/asset.md) into CSV
+
+Consider the following Python snippet:
+
+```python
+from kolena.annotation import Keypoints
+from kolena.io import dataframe_to_csv
+
+df = pd.read_csv(f"s3://kolena-public-examples/300-W/raw/300-W.csv", storage_options={"anon": True})
+df["face"] = df["points"].apply(lambda points: Keypoints(points=json.loads(points)))
+dataframe_to_csv(df, "processed.csv")
+```
+
+#### To use compound metrics on the fly
```

Comment:
A bit abrupt to jump to compound metrics without preamble, and under the context of datapoint not results.

Reply:
this is a good point, do we want move this under `quality-standard/#metrics`? but we don't talk anything regarding ingestion in that page

Reply:
This is good as a starting point but +1 to @ly-kolena's comment that this transition is abrupt. Needs an introductory sentence explaining what compound metrics are, and why/when you would use them.